### PR TITLE
Add logic to every endpoint to be able to use the new permissions based authorization system

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR is not related to any issue. It adds the required permissions to the 4.0 routes.

<hr>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Make sure the TO and TP tests pass. Manually test the 4.0 routes (as many as you can) with different roles to make sure that only a role (associated with a user) with the required permissions can access those routes.
## If this is a bugfix, which Traffic Control versions contained the bug?

- master

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by ocket8888 — CHANGES_REQUESTED** [review]
We shouldn't remove `/capabilities` with this PR, it's an unrelated body of work, and one that constitutes more than is being done here from what I can tell. I opened #6319 to track it.

File: traffic_ops_db/test/docker/run-db-test.sh [hunk 1]
```diff
@@ -99,12 +99,6 @@ fi
 
 ./db/admin --env=production upgrade || { echo "DB upgrade failed!"; exit 1; }
 
-if ! ./db/admin -env=production load_schema ||
-  ! ./db/admin -env=production load_schema; then
-  echo 'Could not re-run create_tables.sql!'
-  exit 1
-fi;
-
```
Comment (zrhoffman) [inline traffic_ops_db/test/docker/run-db-test.sh:107]: As pointed out to me by @ocket8888, this means #4984 becomes reproducible again, which would be fixed by fixing #6106.

Comment (ocket8888) [reply traffic_ops_db/test/docker/run-db-test.sh:107]: I'm good with that, if you are.

**Review by ocket8888 — APPROVED** [review]
Verified manually that a user without enough Permissions to use an endpoint now gets rejected with an error describing said Permissions when attempting to use it anyway - except a user with the `admin` Role, which is always granted access, regardless of Permissions.

File: traffic_ops_db/test/docker/run-db-test.sh (hunk 1 above)
Comment (zrhoffman) [reply traffic_ops_db/test/docker/run-db-test.sh:107]: Yep that's fine with me, as long as #6106 gets fixed at some point.

File: traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql [hunk 2]
```diff
@@ -0,0 +1,138 @@
+/*
```
Comment (rawlinp) [inline traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: I'm not sure if it's a big deal or not, but this file has a couple extra digits in the timestamp compared to prior timestamps:
```
2021100114200000_remove_user_role.down.sql
2021100114200000_remove_user_role.up.sql
202110191500000000_add_permissions.down.sql
202110191500000000_add_permissions.up.sql
```
was that intentional?

Comment (ocket8888) [reply traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: @srijeet0406 Did you generate these with `admin` or make the filenames yourself?

Comment (zrhoffman) [reply traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: The extra digits should have made the *todb-tests* action fail (was it even triggered on this PR?):
https://github.com/apache/trafficcontrol/blob/631d8793eb7e83f9db2624f1cc0723f9f2d4ff28/.github/actions/todb-tests/entrypoint.sh#L86-L96

Comment (srijeet0406) [reply traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: I named them myself, I wasn't even aware that I could generate them using `admin`. I'll create a follow up PR.

Comment (ocket8888) [reply traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: > was it even triggered on this PR?

Yeah, this is the PR where we had to stop checking for being able to run `create_tables` twice, remember? The problem is that `VIOLATING_FILES` isn't being set properly, the set is empty after that `sed` call:

```console
$ ls | sort | cut -d _ -f 1 | sed -n -e '/2020061622101648/,$p' 
$
```

Comment (zrhoffman) [reply traffic_ops/app/db/migrations/202110191500000000_add_permissions.down.sql:1]: Weird that it doesn't show up in the list of actions run above. It shows up when [searching runs by branch `CDN-13621`](https://github.com/apache/trafficcontrol/actions/workflows/traffic.ops.database.yml?query=branch:CDN-13621), though.

**Pull request merged**